clients/metadata/device.go

```diff
@@ -74,35 +76,22 @@ type DeviceClient interface {
 }
 
 type deviceRestClient struct {
-	url      string
-	endpoint clients.Endpointer
+	client interfaces.RestClientBuilder
```

Comment:
Match field name to contract name; i.e. RestClientBuilder type -> restClientBuilder (or builder) field name.

Commented once; may apply elsewhere.